app/models/custom_field/hierarchy/item.rb

```diff
@@ -37,4 +37,8 @@ class CustomField::Hierarchy::Item < ApplicationRecord
   scope :including_children, -> { includes(children: :children) }
 
   def to_s = short.nil? ? label : "#{label} (#{short})"
+
+  def ancestry_path
+    self_and_ancestors.filter_map(&:to_s).reverse.join(" / ")
```

Comment:
🟡 Let's extract the gem methods to the persistence layer, so, instead of calling `self_and_ancestors` directly from model, I'd say we introduce a simple method in the service wrapping it, like we did with descendants:

```ruby
      # Gets all descendant nodes in a tree starting from the item/node.
      # @param item [CustomField::Hierarchy::Item] the node
      # @param include_self [Boolean] flag
      # @return [Success(Array<CustomField::Hierarchy::Item>)]
      def get_descendants(item:, include_self: true)
        if include_self
          Success(item.self_and_descendants)
        else
          Success(item.descendants)
        end
      end
```

Reply:
Hmmm... I don't see the advantage in this case. It would be the same as extracting a scope to an external object.

This literally is asking the node what is its path. =/

Reply:
I'd describe the difference as follows:

The service is the abstraction of the persistence from the domain layer. In this case, you'd be right, finding an item by ID is also a direct access.
(Now the famous) BUT: to be pragmatic, I do not intend to root EVERYTHING through it, but actually everything closure tree related. This way, we could replace the dependency without touching dozens of code files, but just the one. 

WDYT?

Reply:
I agree with you. In this case, we would need to change the actual model anyway so moot point. =)

Reply:
Yeah, eyes opened now.... The model is actually more on persistence layer, sadly :D Makes no sense, using the service in here, as actually the service is using the model and the closure tree methods on it. So, If we would have imports, we would have a circular dependency even.

You are right, I'm wrong